mlir/test/Dialect/Vector/canonicalize.mlir

```diff
@@ -1344,6 +1344,20 @@ func.func @extract_strided_broadcast4(%arg0: f32) -> vector<1x4xf32> {
 
 // -----
 
+// CHECK-LABEL: func @extract_strided_broadcast5
+//  CHECK-SAME: (%[[ARG:.+]]: vector<2x1xf32>)
+//       CHECK: %[[V:.+]] = vector.broadcast %[[ARG]] : vector<2x1xf32> to vector<2x4xf32>
+//       CHECK: return %[[V]]
+func.func @extract_strided_broadcast5(%arg0: vector<2x1xf32>) -> vector<2x4xf32> {
```

Comment:
I see that you are simply following the pre-existing naming convention here, which is the most natural thing to do. Sadly, this function name does not reveal what makes this case unique. Could you add a comment to explain? IIUC, it's the fact that the _trailing_ unit dim is broadcasted?

Reply:
Done. This tests the case where a dimension is both broadcasted and sliced, which just happens to be the trailing dimension.

I also added another dimension of size 2 that is untouched so that this test would fail without this change, since otherwise we fall into the scalar case of the original code.